README.md

```diff
@@ -1,34 +1,64 @@
+The future versions of this tool will be more generalized, allowing users to build a wider range of Ubuntu images, including all future releases.
 
+## Getting started  
 
-## Requirements
+To get started, you will need Ubuntu 18.04 (Bionic Beaver) as a minimum platform requirement. However, it is recommended to use Ubuntu 20.04 (Focal Fossa) or newer. All necessary third-party packages are available in the Ubuntu archive.
 
-Ubuntu 18.04 (Bionic Beaver) is the minimum platform requirement, but Ubuntu 20.04 (Focal Fossa) or newer is recommended. All required third party packages are available in the Ubuntu archive.
+### Build snap-based images
 
+To build a snap-based image with ubuntu-image, you need a [model assertion](https://ubuntu.com/core/docs/reference/assertions/model?_ga=2.79789975.141992633.1711553306-121934327.1710189436). A model assertion is a YAML file that describes a particular combination of core, kernel, and gadget snaps, along with other declarations, signed with a digital signature asserting its authenticity. The ubuntu-image command only requires the path to this model assertion to build snap-based images.
+
+You can build snap-based images with ubuntu-image using the following command:
+
+```bash
+ubuntu-image snap model.assertion
+```
+
+You can find more information about building snap-based images using ubuntu-images [here](https://discourse.ubuntu.com/t/using-ubuntu-image/35216#heading--ubuntu-image). To build an image with custom snaps, visit [Build an image with custom snaps](https://discourse.ubuntu.com/t/build-an-image-with-custom-snaps/19809).
+
+### Build classical images
+
+Classical images are built from a local [gadget tree](https://github.com/snapcore/pc-gadget) path. The [gadget tree](https://github.com/snapcore/pc-gadget) is a primed [gadget snap](https://snapcraft.io/docs/the-gadget-snap) containing a `gadget.yaml` file in the meta directory and all the necessary bootloader gadget bits built. You can prepare a [gadget tree](https://github.com/snapcore/pc-gadget) by fetching a specially customised [gadget snap](https://snapcraft.io/docs/the-gadget-snap) source and running snapcraft prime on it. The resulting tree will be stored in the prime/ directory. 
+
+The `gadget.yaml` file contains important information about the volumes that will be created. You can think of these volumes as disk images. The file also specifies the structures that will be included within each volume. Structures define the layout of the volume, including partitions, Master Boot Records, and any other relevant content. Also, the `gadget.yaml` file indicates whether a bootloader will be included in the volume and, if so, the type of bootloader it will be.
+
+You can build classical images with ubuntu-image using the following command:
+
+```bash
+ubuntu-image classic image_definition.yaml
+```
+
+For more information on how to build a gadget snap, visit [Building a gadget snap](https://ubuntu.com/core/docs/gadget-building).
```

Comment:
Considering we have a section on building classical images, it seems strange that snap-based images don't have their own section.

How about having just a generic section called "Building images", which would only include very basic information and then links to instructions on how to do the actual building?